src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java

```diff
@@ -34,80 +35,173 @@ public class JSONCodecRegistry {
    *
    * <p>
    *
-   * @param targetCQLType CQL type of the target column we want to write to.
-   * @param javaValue Java object that we want to write to the column.
-   * @return Optional of the codec that can convert the Java object into the object expected by the
-   *     CQL driver. If no codec is found, the optional is empty.
+   * @param table {@link TableMetadata} to find the column definition in
+   * @param column {@link CqlIdentifier} for the column we want to get the codec for.
+   * @param value The value to be written to the column
    * @param <JavaT> Type of the Java object we want to convert.
    * @param <CqlT> Type fo the Java object the CQL driver expects.
+   * @return The {@link JSONCodec} that can convert the value to the expected type for the column,
+   *     or an exception if the codec cannot be found.
+   * @throws UnknownColumnException If the column is not found in the table.
+   * @throws MissingJSONCodecException If no codec is found for the column and type of the value.
    */
-  public static <JavaT, CqlT> Optional<JSONCodec<JavaT, CqlT>> codecFor(
-      DataType targetCQLType, JavaT javaValue) {
-
-    return Optional.ofNullable(
-        JSONCodec.unchecked(
-            CODECS.stream()
-                .filter(codec -> codec.test(targetCQLType, javaValue))
-                .findFirst()
-                .orElse(null)));
+  public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToCQL(
+      TableMetadata table, CqlIdentifier column, Object value)
+      throws UnknownColumnException, MissingJSONCodecException {
+
+    Preconditions.checkNotNull(table, "table must not be null");
```

Comment:
These checks will create runtime exceptions. Need to be handled with error codes.